plugins/gatsby-plugin-generate-doc-json/utils/handlers.js

```diff
@@ -139,6 +157,18 @@ module.exports = {
       [u('text', '\u00A0')]
     );
   },
+  ImageSizing: (h, node) => {
+    const style = stripNulls({
+      height: findAttribute('height', node),
+      width: findAttribute('width', node),
+      verticalAlign: findAttribute('verticalAlign', node),
+    });
+
+    const [image] = node.children;
+    set(image, 'data.style', compileStyleObject(style));
+
+    return one(h, image, node);
```

Comment:
tl;dr; you might need to make the `parent` the `node.parent` instead of the `ImageSizing` node itself.

**Why?**

Now that I look at this one more time, I'm realizing that those `rehype` plugins that handle image sizing and inline images have a bit of an order dependency. If you look at the [`image-sizing` plugin](https://github.com/newrelic/docs-website/blob/develop/rehype-plugins/image-sizing/index.js), you'll notice that the `ImageSizing` removes itself from the tree and instead [replaces itself with the image node](https://github.com/newrelic/docs-website/blob/develop/rehype-plugins/image-sizing/index.js#L34). This means there will be no additional wrapper. Because of this change, the image node's parent now becomes the `ImageSizing` node's parent.

This is important, because the [`gatsby-inline-images`](https://github.com/newrelic/docs-website/blob/develop/rehype-plugins/gatsby-inline-images/index.js) plugin [runs after](https://github.com/newrelic/docs-website/blob/develop/gatsby-config.js#L231-L237) the `image-sizing` plugin runs. This is important because the `image-sizing` plugin checks to see if its parent is a `paragraph` and has the image as its only child (much like you're doing here).

If the image handler checks to see if its parent is a `paragraph`, but the `ImageSizing` node is not "removed", then you may not be able to detect block level images that use `ImageSizing` (of which there are definitely pages that do).

In other words, you won't be able to properly handle this MDX:

```mdx
Some paragraph

<ImageSizing width="20px">
  ![icon.png](./images/icon.png)
</ImageSizing>

More stuff
```

The space in between those paragraphs are what cause the AST to wrap a paragraph around the `ImageSizing` and `image` markdown. To ensure you can detect the above as a block image, you'll want the image's parent to be the `ImageSizing` parent. Does that make sense?